### PR TITLE
os/board/rtl8720e: Fix warnings

### PR DESCRIPTION
**Pull request by kaikaikaime**
Add type casting
    Remove unused variables
    Initialise variables
    Update define condition for amebalite

### CODE REVIEW

File: apps/examples/board_specific_demo/ameba_wifi_csi_demo/example_wifi_csi.c [hunk 1]
```diff
@@ -95,7 +94,7 @@ int wificsi_main(int argc, char *argv[])
 	act_param.act = 0;  /* csi en */
 	act_param.enable = 1;
 	wifi_csi_config(&act_param);
-	csi_buf = malloc(csi_data_len);
+	csi_buf = (unsigned char*)malloc(csi_data_len);
```
Comment (sunghan-chang) [inline apps/examples/board_specific_demo/ameba_wifi_csi_demo/example_wifi_csi.c:97]: Could you add a space between type and pointer?
```
(unsigned char *)malloc
```

Comment (thapav) [inline apps/examples/board_specific_demo/ameba_wifi_csi_demo/example_wifi_csi.c:97]: Please change to (unsigned char *)

File: os/arch/arm/src/amebalite/amebalite_timerisr.c [hunk 2]
```diff
@@ -85,7 +85,7 @@
  */
 extern uint32_t SystemCoreClock; 
 
-#define SYSTICK_RELOAD ((SystemCoreClock / CLK_TCK) - 1)
+#define SYSTICK_RELOAD ((150000000 / CLK_TCK) - 1)
```
Comment (thapav) [inline os/arch/arm/src/amebalite/amebalite_timerisr.c:88]: Hardcoding is not correct, kindly revert this

File: os/board/rtl8720e/src/component/bluetooth/api/rtk_bt_common.c [hunk 3]
```diff
@@ -748,8 +748,8 @@ uint16_t rtk_bt_set_evt_cb_direct_calling(uint8_t group, uint32_t evt_bit_mask)
 
 	direct_call_t.group = group;
 	direct_call_t.evt_bit_mask = evt_bit_mask;
-	ret = bt_ipc_api_host_message_send(RTK_BT_IPC_COMMON, RTK_BT_ACT_IPC_ENABLE_DIRECT_CALL,
-										&direct_call_t, sizeof(rtk_bt_direct_call_t));
+	ret = bt_ipc_api_host_message_send(RTK_BT_IPC_COMMON, RTK_BT_ACT_IPC_ENABLE_DIRECT_CALL, 
+									(uint8_t*)&direct_call_t, sizeof(rtk_bt_direct_call_t));
```
Comment (thapav) [inline os/board/rtl8720e/src/component/bluetooth/api/rtk_bt_common.c:752]: Add space between the type & asterisk for all the type casts

**Review by thapav — COMMENTED** [review]
Please add space between the type & asterisk for all the type casts
(type *)

File: os/board/rtl8720e/src/component/bluetooth/api/rtk_bt_common.c (hunk 3 above)
Comment (kaikaikaime) [reply os/board/rtl8720e/src/component/bluetooth/api/rtk_bt_common.c:752]: I will make the changes again

Comment (sunghan-chang) [reply os/board/rtl8720e/src/component/bluetooth/api/rtk_bt_common.c:752]: @kaikaikaime @thapav We don't recommend to follow TizenRT coding rules in the BSP (chipset vendor codes). @kaikaikaime You don't need to follow it here. You can decide and let us know about modification (will do or not).

Comment (sunghan-chang) [reply os/board/rtl8720e/src/component/bluetooth/api/rtk_bt_common.c:752]: @kaikaikaime Could you let me know your plan? Will you keep current (merge current) or will make changes (wait your changes)?

Comment (kaikaikaime) [reply os/board/rtl8720e/src/component/bluetooth/api/rtk_bt_common.c:752]: Hi @sunghan-chang, I may need to discuss with my colleague on this, I will get back to you latest at the end of the day.

**Comment by kaikaikaime** [comment]
Hi, I have made the necessary changes mentioned above and updates

File: build/configs/rtl8720e/hello/defconfig [hunk 4]
```diff
@@ -127,8 +127,8 @@ CONFIG_ARCH_CHIP_RTL8720E=y
 CONFIG_RTL8720E_UART=y
 CONFIG_RTL8720E_UART0=y
 CONFIG_RTL8720E_UART1=y
-CONFIG_RTL8720E_UART2=y
-CONFIG_RTL8720E_UART3=y
+#CONFIG_RTL8720E_UART2=y
+#CONFIG_RTL8720E_UART3=y
```
Comment (sunghan-chang) [inline build/configs/rtl8720e/hello/defconfig:131]: Same as previous. Please check my previous comment. This is not allowed.

File: apps/examples/board_specific_demo/ameba_secure_storage_demo/secure_storage.c [hunk 5]
```diff
@@ -141,14 +141,14 @@ int sstorage_main(int argc, char *argv[])
 	}
 
 	printf("Read Raw data from Slot: %d, Area A\n", slot_index);
-	if (!rtl_ss_flash_read(address, length, rwbuf, 1)) {
+	if (!rtl_ss_flash_read(address, length, (uint8_t *) rwbuf, 1)) {
```
[the remark this answers is not in the document]
Comment (sunghan-chang) [reply apps/examples/board_specific_demo/ameba_secure_storage_demo/secure_storage.c:144]: Please check my previous comment.

**Comment by kaikaikaime** [comment]
Hi, 
Thank you for your patience, I indeed left out some.

File: build/configs/rtl8720e/hello/defconfig [hunk 6]
```diff
@@ -127,8 +127,8 @@ CONFIG_ARCH_CHIP_RTL8720E=y
 CONFIG_RTL8720E_UART=y
 CONFIG_RTL8720E_UART0=y
 CONFIG_RTL8720E_UART1=y
-CONFIG_RTL8720E_UART2=y
-CONFIG_RTL8720E_UART3=y
+# CONFIG_RTL8720E_UART2 is not set
+# CONFIG_RTL8720E_UART3 is not set
```
Comment (sunghan-chang) [inline build/configs/rtl8720e/hello/defconfig:131]: This is still modified manually. When I turn them off using menuconfig, other configs are modified together due to dependencies. Please update this using menuconfig and then verify them with changed.

Comment (kaikaikaime) [reply build/configs/rtl8720e/hello/defconfig:135]: I have pushed, and fixed some defconfig files, the hello config now matches the .config of the generated menuconfig

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**